rustls/src/error.rs

```diff
@@ -262,6 +262,19 @@ pub enum CertificateError {
     Other(Arc<dyn StdError + Send + Sync>),
 }
 
+impl From<CertificateError> for AlertDescription {
+    fn from(e: CertificateError) -> Self {
+        use CertificateError::*;
+        match e {
+            BadEncoding | UnhandledCriticalExtension | NotValidForName => Self::BadCertificate,
+            Expired | NotValidYet => Self::CertificateExpired,
```

Comment:
Do you think it's potentially confusing to turn `NotValidYet` into `CertificateExpired`? OpenSSL turns `NotValidYet` into `BadCertificate`, BoringSSL turns it into `CertificateExpired` like you have here.

Reply:
Following the [RFC](https://www.rfc-editor.org/rfc/rfc5246#section-7.2.2)

>    certificate_expired
      A certificate has expired or is not currently valid.

Reply:
Ah, right!